addons/website_sale/models/product.py

```diff
@@ -179,17 +182,18 @@ def _is_quick_add_to_cart_possible(self, parent_combination=None):
 
         if not self._is_add_to_cart_possible(parent_combination):
             return False
-        if len(self._get_possible_variants(parent_combination)) != 1:
-            return False
-        if self._has_no_variant_attributes():
-            return False
-        if self.has_dynamic_attributes():
-            return False
-        if self._has_is_custom_values():
-            return False
-        if self.optional_product_ids.filtered(lambda p: p._is_add_to_cart_possible(self._get_first_possible_combination())):
+        gen = self._get_possible_combinations(parent_combination)
+        first_possible_combination = next(gen)
+        try:
+            next(gen)
+            # If it didn't raise, there are at least 2 possible combinations.
             return False
-        return True
+        except StopIteration:
+            if self._has_is_custom_values():
```

Comment:
```python
except YourException:
   pass

if self...
```

Reply:
btw

in your case, you can give a default value a second parameter of next to avoid this StopIteration

Reply:
Same as above, I just need to test if the iteration works here, I don't want to return its result.

Reply:
Following our discussion, using `next(gen, False) is not False` instead